include/linker/devicetree_regions.h

```diff
@@ -35,25 +35,29 @@
 #define LINKER_DT_NODE_REGION_NAME(node_id) \
 	DT_PROP_OR(node_id, zephyr_linker_region, DT_NODE_PATH(node_id))
 
-/* Declare a memory region */
-#define _REGION_DECLARE(name, attr, node) name(attr) : \
-	ORIGIN = DT_REG_ADDR(node),		       \
-	LENGTH = DT_REG_SIZE(node)
+/** @cond INTERNAL_HIDDEN */
 
 /**
- * @brief Generate a linker memory region from a devicetree node
- *
- * If @p node_id refers to a node with status "okay", then this declares
- * a linker memory region named @p name with attributes from @p attr.
+ * @brief Declare a memory region
  *
- * Otherwise, it doesn't expand to anything.
+ * @param node_id devicetree node identifier
+ * @param attr region attributes
+ */
+#define _REGION_DECLARE(node_id, attr)		    \
+	LINKER_DT_NODE_REGION_NAME(node_id)(attr) : \
```

Comment:
you would probably want to consider removing `attr` at all (that at least for the GNU linker is optional). The problem is that when the MPU is present the actual attribute of the region (in terms of RWX) is defined by the MPU and by the attributes that we are using to mark the region and not by the linker.

If you think to introduce the `mpu-attr` we could think of obtaining this `attr` from the `mpu-attr` but that is hard.

Reply:
That can be done in future PR's if needed. This is purely about updating where the names are coming from, I don't want to potentially introduce other problems here.

Reply:
This PR is large enough as it it

Reply:
Ok, I'm only concerned that all the future needed changes (`mpu-attr`, the `FOR_EACH_..` macro, this fix on the `_REGION_DECLARE`, etc...) will require deep changes to the current state of this PR.

I have no objections on this PR if this is considered a first step.